

diff --git a/drivers/edac/igen6_edac.c b/drivers/edac/igen6_edac.c
index 405f3dad73aa..ae8c1afe672c 100644
--- a/drivers/edac/igen6_edac.c
+++ b/drivers/edac/igen6_edac.c
@@ -27,6 +27,7 @@
 #include <linux/sched/clock.h>
 #include <asm/cpu_device_id.h>
 #include <asm/intel-family.h>
+#include <asm/mach_traps.h>
 
 #include "edac_mc.h"
 #include "edac_module.h"
@@ -469,6 +470,7 @@
 static int ecclog_nmi_handler(unsigned int cmd, struct pt_regs *regs)
 {
 	u64 delta, ecclog;
+	unsigned char nmi_reason;
 
 	raw_spin_lock(&ecclog_lock);
 
@@ -492,6 +494,12 @@
 		irq_work_queue(&ecclog_irq_work);
 	last_handle_jiffies = jiffies;
 
+	/* Toggle the SERR clear bit in order to re-enable the SERR error for the next time */
+	nmi_reason = inb(NMI_REASON_PORT);
+	nmi_reason &= NMI_REASON_CLEAR_MASK;
+	outb(nmi_reason | NMI_REASON_CLEAR_SERR, NMI_REASON_PORT);
+	outb(nmi_reason & (~NMI_REASON_CLEAR_SERR), NMI_REASON_PORT);
+
 	raw_spin_unlock(&ecclog_lock);
 
 	return NMI_HANDLED;
